Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -21,14 +44,26 @@ internal struct DefaultOperationExecutor: OperationExecutor {
     internal func execute<T: Operation>(_ operation: T,
                                         client: MongoClient,
                                         session: ClientSession?) throws -> T.OperationResult {
-        // if a session was provided, use its underlying connection
-        if let session = session {
-            let conn = try session.getConnection(forUseWith: client)
+        switch T.connectionUsage {
+        case .steals:
+            // don't return the connection to the pool, as the operation will handle it
+            let conn = try session?.getConnection(forUseWith: client) ?? client.connectionPool.checkOut()
             return try operation.execute(using: conn, session: session)
-        }
-        // otherwise use a new connection from the pool
-        return try client.connectionPool.withConnection { conn in
-            try operation.execute(using: conn, session: nil)
+        case .ignores:
+            // we won't use the connection so pass in a dummy. initialization always succeeds with this bit pattern.
+            // swiftlint:disable:next force_unwrapping
+            let dummy = Connection(OpaquePointer(bitPattern: 1)!)
```

Comment:
I think it would be better if we modified the `Operation` protocol method to take in a `Connection?` rather than just passing in a null pointer.

Reply:
we'd have to unwrap the `Connection` in all of the operations that actually use it then, do you think that's preferable?

Reply:
That's also not great, but it feels a little off to side-step the compiler like we're currently doing here. I don't feel super strongly about this though, so if you think it's fine, then it's fine with me.

Reply:
one alternative I thought of:
we make the `connectionUsage` property non-static, and make `.ignores` something like `.owns(Connection)` instead.

then whenever it's that case, we just pass in the already-owned `Connection` to the `execute` method instead.

so e.g. `NextOperation` would be `.owns` and use the cursor's source connection in that place.

thoughts?

Reply:
Yeah I think that sounds good. That's actually how I currently have it designed in the Rust driver, too (though we went that path mostly for Rust specific reasons).

Reply:
updated to that approach